src/main/java/racingcar/domain/Car.java

```diff
@@ -0,0 +1,36 @@
+package racingcar.domain;
+
+
+import racingcar.constant.DomainConstant;
+import racingcar.strategy.RacingCarMoveStrategy;
+
+public class Car {
+    private final CarName name;
+    private final Lap lap;
+
+    public Car(String name) {
+        this.name = new CarName(name);
+        this.lap = new Lap(DomainConstant.LAP_MIN_SIZE.getValue());
+    }
+
+    private Car(CarName name, Lap lap) {
+        this.name = name;
+        this.lap = lap;
+    }
+
+    public Car move(RacingCarMoveStrategy racingCarMoveStrategy) {
```

Comment:
클래스의 상태를 불변으로 설계하신 건가요? 
이런 방법도 좋은 것 같습니다 :)

Reply:
넵! 맞습니다 :)

Reply:
전진하게 될때 새로운 객체를 생성하는 비용이 발생하게 될거같은데
불변으로 만드신 이유가 궁금합니다.

Reply:
불변 객체를 생성하는데 비용이 발생하는 것에 대해 크게 걱정하지 않아도 됩니다.
아래는 오라클 공식 문서의 원문을 가져온 내용 중 일부인데 참고하시면 좋을 것 같습니다 :)
> Programmers are often reluctant to employ immutable objects, because they worry about the cost of creating a new object as opposed to updating an object in place. The impact of object creation is often overestimated, and can be offset by some of the efficiencies associated with immutable objects. These include decreased overhead due to garbage collection, and the elimination of code needed to protect mutable objects from corruption.